client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+	recordToWrite := pdsClient.WriteRecordRequest{
+		Data: data,
+		Metadata: pdsClient.Meta{
+			Type:     recordType,
+			WriterID: c.E3dbPDSClient.ClientID,
+			UserID:   c.E3dbPDSClient.ClientID,
+			Plain:    plain,
+		},
+	}
+	encryptedRecord, err := c.E3dbPDSClient.EncryptRecord(ctx, recordToWrite)
+	if err != nil {
+		return nil, err
+	}
+	recordSecret, err := c.E3dbPDSClient.WriteRecord(ctx, encryptedRecord)
+	if err != nil {
+		return nil, err
+	}
+	return recordSecret, nil
+}
+
+func (c *ToznySDKV3) ShareRecordWithGroup(ctx context.Context, recordType string, group *storageClient.Group) error {
+	keyReq := pdsClient.GetOrCreateAccessKeyRequest{
+		WriterID:   c.E3dbPDSClient.ClientID,
+		UserID:     c.E3dbPDSClient.ClientID,
+		ReaderID:   c.E3dbPDSClient.ClientID,
+		RecordType: recordType,
+	}
+	keyResp, err := c.E3dbPDSClient.GetOrCreateAccessKey(ctx, keyReq)
+	if err != nil {
+		return err
+	}
+	encryptionKeys := e3dbClients.EncryptionKeys{
+		Public: e3dbClients.Key{
+			Type:     e3dbClients.DefaultEncryptionKeyType,
+			Material: c.StorageClient.EncryptionKeys.Public.Material,
+		},
+		Private: c.StorageClient.EncryptionKeys.Private,
+	}
+	wrappedAK, err := e3dbClients.EncryptAccessKey(keyResp, encryptionKeys)
+	if err != nil {
+		return err
+	}
+	accessKeyReq := storageClient.GroupAccessKeyRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: wrappedAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, encryptedGroupAK, err := c.StorageClient.CreateGroupAccessKey(ctx, accessKeyReq)
+	if err != nil {
+		return err
+	}
+	// share secret with group
+	secretShareReq := storageClient.ShareGroupRecordRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: encryptedGroupAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, err = c.StorageClient.ShareRecordWithGroup(ctx, secretShareReq)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetOrCreateNamespace creates the group for the namespace if it doesn't exist and returns the Group
+func (c *ToznySDKV3) GetOrCreateNamespace(ctx context.Context, realmName string, namespace string) (*storageClient.Group, error) {
+	var group *storageClient.Group
+	groupName := fmt.Sprintf("tozny.secret.%s.%s", realmName, namespace)
+	listRequest := storageClient.ListGroupsRequest{
+		ClientID:   uuid.MustParse(c.StorageClient.ClientID),
+		GroupNames: []string{groupName},
+	}
+	responseList, err := c.StorageClient.ListGroups(ctx, listRequest)
+	if err != nil {
+		return nil, err
+	}
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	if err != nil {
+		return nil, err
+	}
+	eak, err := e3dbClients.EncryptPrivateKey(encryptionKeyPair.Private, c.StorageClient.EncryptionKeys)
+	if err != nil {
+		return nil, err
+	}
+	if len(responseList.Groups) < 1 {
+		// create group
+		groupReq := storageClient.CreateGroupRequest{
```

Comment:
nit: I won't let you save more time by leaving off `uest` because now you have to read this comment. But you can always ignore me, this is a nit and not a blocker.